### PR TITLE
[SYCL][Fusion] Test fusion of kernels with different ND-ranges

### PR DESCRIPTION
**Pull request by victor-eds**
Check that kernel fusion works when fusing kernels with different ND-ranges (different global sizes and dimensions).

Implementation: intel/llvm#8209

Signed-off-by: Victor Perez <victor.perez@codeplay.com>

### CODE REVIEW

**Comment by victor-eds** [comment]
\verify with https://github.com/intel/llvm/pull/8209

File: SYCL/KernelFusion/different_nd_ranges.cpp [hunk 1]
```diff
@@ -0,0 +1,246 @@
...
+                      return std::equal(LHS.begin(), LHS.end(), RHS.begin(),
+                                        [](const auto &LHS, const auto &RHS) {
+                                          return all(LHS == RHS);
+                                        });
+                    }) &&
+         "COMPUTATION ERROR");
+}
+
+int main() {
+  test({RangeDesc{10}, RangeDesc{20}});
```
Comment (sommerlukas) [inline SYCL/KernelFusion/different_nd_ranges.cpp:238]: Can you please add a short comment on each `test` line explaining the scenario?

File: SYCL/KernelFusion/different_nd_ranges.cpp [hunk 2]
```diff
@@ -0,0 +1,246 @@
+// RUN: %clangxx -fsycl -fsycl-targets=%sycl_triple %s -o %t.out
+// RUN: %CPU_RUN_PLACEHOLDER %t.out
+// RUN: %GPU_RUN_PLACEHOLDER %t.out
+// UNSUPPORTED: cuda || hip
+// REQUIRES: fusion
+
+// Test complete fusion without any internalization
```
Comment (sommerlukas) [inline SYCL/KernelFusion/different_nd_ranges.cpp]: Update to match the actual purpose of the test.

**Review by sommerlukas — COMMENTED** [review]
Just a few documentation nits.

File: SYCL/KernelFusion/different_nd_ranges.cpp [hunk 3]
```diff
@@ -0,0 +1,265 @@
...
+  // Three 1-D kernels with different global sizes and specified (equal) local
+  // size.
+  const auto R5 = {5ul};
+  test({RangeDesc{{10}, R5}, RangeDesc{{20}, R5}, RangeDesc{{30}, R5}});
+
+  // Two 1-D kernels with different global sizes and a 2-D kernel with more
+  // work-items and specified (equal) local size.
+  test({RangeDesc{{10}, R2}, RangeDesc{{20}, R2}, RangeDesc{{10, 10}, {2, 1}}});
+
+  // Three 2-D kernels with different global sizes..
```
Comment (sommerlukas) [inline SYCL/KernelFusion/different_nd_ranges.cpp]:
```suggestion
  // Three 2-D kernels with different global sizes.
```

**Review by sommerlukas — APPROVED** [review]

**Review by steffenlarsen — APPROVED** [review]

**Comment by steffenlarsen** [comment]
Validation of https://github.com/intel/llvm/pull/8209 passes these tests.

**Pull request merged**